segmentation/utils/signal_extraction_test.py

```diff
@@ -4,17 +4,54 @@
 from segmentation.utils import signal_extraction
 
 
+def test_positive_pixels_extraction():
+    # this function tests the functionality of positive pixels extraction
+    # where we count the number of non-zero values for each channel
+
+    # create a series of constant arrays
```

Comment:
Lets use the new synthetic testing functions here instead. 

Once you've added the option for a threshold, we can ensure that with a threshold of 0 all pixels are returned, and then use a different threshold that only applies to the signal from one channel and not the other

Reply:
So looking back I realize it wasn't entirely clear as to how we should use a threshold that only applies to signal from one channel. Currently, synthetic_spatial_datagen labels signal with constant values regardless of which channel is used (currently set to 10). So using a set threshold for every channel means that we either get all or nothing from both membrane and nuclear.

Did you by any chance mean that the user should be allowed to specify a different threshold for each channel (ex. 5 for nuclear, and 8 for membrane)? Or would ideally the channel-generation be more diverse in labeling signal (ex. different signal values for nuclear and membrane)?

Reply:
The latter. so we can add a parameter to the membrane and nuclear synthetic data functions to specify what the constant value is that gets used, and set a different one for each to test this specific part. 

Unless you have a different idea for a simpler way to test it, in which case go ahead and do that instead! Whatever you think is best